src/main/java/com/powsybl/openloadflow/ac/outerloop/TransformerVoltageControlOuterLoop.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.ac.outerloop;
+
+/*import com.powsybl.openloadflow.equations.Equation;
+import com.powsybl.openloadflow.equations.EquationType;
+import com.powsybl.openloadflow.equations.Variable;
+import com.powsybl.openloadflow.equations.VariableType;
+import com.powsybl.openloadflow.network.LfBranch;
+import com.powsybl.openloadflow.network.PhaseControl;
+import com.powsybl.openloadflow.network.PiModel;*/
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+public class TransformerVoltageControlOuterLoop implements OuterLoop {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TransformerVoltageControlOuterLoop.class);
+
+    @Override
+    public String getType() {
+        return "Transformer voltage control";
+    }
+
+    @Override
+    public OuterLoopStatus check(OuterLoopContext context) {
```

Comment:
I don't known if it is clear for you what has to be done here but here is a raw extract (to translate) of what is done in H2:

Articulation des 3 load flows pour la simulations des regleurs en charges
 * Pour rappel:
 * -LF1 = gpesTHT PV, grpes HT PV, regleurs fixes
 * -LF2 = gpesTHT PV, grpes HT PQ, regleurs variables
 * -LF3 = gpesTHT PV, grpes HT PV, regleurs fixes (apres discretisation)

Reply:
I am not sure that we have to follow these 3 steps here. I think that I am going to follow simply what we have done for phase tap changer. It will be easy after that to make a more business and complex outerloop. In the EMF specifications, it is just written:
When calculating the steady state solution, a full Newton Raphson power flow algorithm is to be used, using the following settings: 
•    Q limits shall be respected (also for slack node/swing bus) 
•    Transformer tap adjustment is set to enabled
•    Switched shunt adjustment is set to enabled
Transformer  tap  and  switch  shunt  control  is  used  during  power  flow  calculation  to  enable regulating control during contingency analysis (where the solved system state is used for the base case). In case of non-convergence, switched shunt and transformer tap regulation is disabled and power flow calculation is started again.

Reply:
Even if not specified you need to follow these 3 steps in order to emulate a realistics tap changer and generation co ntrol behaviour.

Reply:
Do we have a way to have a generic outerloop that can be customize easily to simulate real world ?

Reply:
Yes just one outerloop and a different behaviour at iteration 0, 1 and 2